File: API/Models/Snippet.cs
using System.ComponentModel.DataAnnotations.Schema;
using API.Models;

namespace API;

public class Snippet
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public string Notes { get; set; }
    public string Creator { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime LastUpdated { get; set; }
    public List<SnippetFilter>? SnippetFilters { get; set; }
    public string AppUserId { get; set; }

}

File: Client/src/app/models/snippet.ts
import { Filter } from './filter';
import { SnippetFilter } from './snippetFilter';

export interface Snippet {
  id?: number;
  title?: string;
  content?: string;
  notes?: string;
  creator?: string;
  dateCreated?: Date;
  lastUpdated?: Date;
  filters?: number[];
  newFilters?: Filter[];
  appUserId?: string;
  snippetFilters?: SnippetFilter[];
}
